eq.yml

```diff
@@ -955,41 +940,119 @@ jobs:
+  - task: Deploy Author-Api
+    params:
+      AWS_ACCESS_KEY_ID: ((preprod_aws_access_key))
+      AWS_SECRET_ACCESS_KEY: ((preprod_aws_secret_key))
+      ANSIBLE_HOST_KEY_CHECKING: False
+      TF_VAR_env: 'preprod'
+      TF_VAR_aws_access_key: '((preprod_aws_access_key))'
+      TF_VAR_aws_secret_key: '((preprod_aws_secret_key))'
+      TF_VAR_vpc_id: '((preprod_vpc_id))'
+      TF_VAR_ecs_cluster_name: 'preprod-eq'
+      TF_VAR_docker_registry: {{docker_registry}}
+      TF_VAR_aws_alb_arn: {{preprod_aws_alb_arn}}
+      TF_VAR_service_name: 'author-api'
+      TF_VAR_container_name: 'eq-author-api'
+      TF_VAR_container_port: 4000
+      TF_VAR_listener_rule_priority: 103
+      TF_VAR_healthcheck_path: '/status'
+      TF_VAR_slack_alert_sns_arn: '((preprod_slack_alert_sns_arn))'
+    config:
+      platform: linux
+      image_resource:
+        type: docker-image
+        source:
+          repository: onsdigital/eq-terraform-build
+      inputs:
+      - name: eq-author-api
+      - name: eq-ecs-deploy
+      run:
+        path: bash
+        args:
+        - -exc
+        - |
+          cd eq-ecs-deploy
+          tfenv install
+
           author_api_tag=$(cat ../eq-author-api/.git/HEAD | xargs echo -n)
-          publisher_tag=$(cat ../eq-publisher/.git/HEAD | xargs echo -n)
 
-          terraform init -backend-config="bucket="concourse-preprod-terraform-state"" -backend-config="key="preprod-author"" -backend-config="region="eu-west-1""
+          terraform init -backend-config="bucket="concourse-preprod-terraform-state"" -backend-config="key="preprod-author""
           terraform apply \
-          -var author_tag=$author_tag \
-          -var author_api_tag=$author_api_tag \
-          -var publisher_tag=$publisher_tag
+          -var container_tag=$author_api_tag \
+          -var 'container_environment_variables="{ \"name\": \"DB_CONNECTION_URI\", \"value\": \"((preprod_author_database_connection_string))\" }"'
```

Comment:
Just a thought that we should probably have noticed earlier.
But passing the full connection string as an env var is really bad practice as it contains the credentials.
These credentials will then be visible in the console.
Generally passing credentials as env var is considered bad as anything on the box can read them.

Reply:
This is currently how runner behaves as well. What is a better way of handling this, mount some volume containing secrets? Or encrypting the environment variables like is done on Travis?

Reply:
Runner loads its secrets at runtime, either from a file that is packaged on deploy for beanstalk and grabs them from s3 when running in ECS

Reply:
Hmm yeah, I see that now https://github.com/ONSdigital/eq-terraform/blob/4d1f977dad9edf9467c0fe68dd7ac6205aa9c101/survey-runner.tf#L42. So how do I progress this? We'd need to make some changes to Author-Api so that it loads in secrets from S3 in a similar way to survey-runner i guess, then I'll need to revisit all of the Docker and terraform config so that it loads the file from somewhere else, if S3 isn't available.

Reply:
There'll be quite a few changes needed to support this. I wonder if it should be done under a different task given that we have always used env vars up until this point?